### PR TITLE
Fix omitting the last point's coordinates in the geometry field

### PR DESCRIPTION
**Pull request by oxidase**
#2546 leg_index is not incremented, so the last point of the last leg was always missing if number of legs > 1

### CODE REVIEW

**Comment by danpat** [comment]
@oxidase nice catch.  Do you have the bandwidth to make a test case for this?

**Comment by oxidase** [comment]
@danpat added a test for geometry overviews. One thing that i don't get in the expectation result is the duplicated point b. Two geometries from assembleGeometry are [1,1;1.000089,1] and  [1.000089,1;1.000089,1;1.000089,0.99991], so the second leg has b as a start node and a middle node.

File: features/support/route.js [hunk 1]
```diff
@@ -163,6 +163,10 @@ module.exports = function () {
         return instructions.legs.map(l => {return zip(l.annotation.duration, l.annotation.distance).map(p => { return p.join(':'); }).join(','); }).join(',');
     };
 
+    this.geometryList = (instructions) => {
+        return instructions.geometry.coordinates.map(p =>{ return p[0] + ',' + p[1]; }).join(';');
```
Comment (TheMarex) [inline features/support/route.js]: Relying on setting `geometries=geojson` here is bad. Easy to be messed up by the calling code. Instead I would suggest decoding the default (polyline) using https://github.com/mapbox/polyline

Also the parameter name `instructions` is misleading: it is a `Route` object.

Comment (oxidase) [inline features/support/route.js]: Ok, changed to decode geometry polyline response and compare decoded strings

**Comment by TheMarex** [comment]
>  One thing that i don't get in the expectation result is the duplicated point b

Hm did you fix that? I observe this in the test case. Going to merge this to make it into `5.2.4`.

**Pull request merged**